module/extension/Director/src/director/remove_task.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include "Director.hpp"
+#include "provider/ProviderGroup.hpp"
+
+namespace module::extension {
+
+    using ::extension::behaviour::commands::DirectorTask;
+
+    void Director::remove_task(const std::shared_ptr<const DirectorTask>& task) {
+
+        // Get the group for this task
+        auto& group = groups[task->type];
+
+        // Check if this task is the currently active task this provider is running
+        if (task == group.active_task) {
+
+            // Remove this task, we are no longer doing it
+            group.active_task = nullptr;
+
+            // Re-evaluate the queue of tasks for this group
+            reevaluate_queue(group);
+
+            // If nothing in the queue updated the active task to a new task we are now idle
+            // That also means we need to remove any subtasks this group had recursively
+            if (group.active_task == nullptr) {
```

Comment:
When does `group.active_task` get updated? Is that something that will be happening in `run_task_pack`? Otherwise if I understand correctly it will set it to `nullptr` and never run the `else`

Reply:
Yeah it should get set in `run_task_pack`, hopefully when the task pack runs it will become the active task